File: C++/Medium/CompileError/robot-bounded-in-circle_8.cpp
class Solution {
public:
    bool isRobotBounded(string instructions) {
        
        int x = 0;
        int y = 0;
        int dir = 0;
        int dx[] = {0, 1, 0, -1};
        int dy[] = {1, 0, -1, 0};
        for (char c: instructions) {
            if (c == 'G') {
                x += dx[dir];
                y += dy[dir];
            } else if (c == 'L') {
                dir = (dir + 3) % 4;
            } else {
                dir = (dir + 1) % 4;
            }
        }
        return (x == 0 && y == 0) |;
            }
        }
        return (x == 0 && y == 0) || dir != 0;
    }
};

File: Rust/Medium/Accepted/robot-bounded-in-circle_8.rs
impl Solution {
    pub fn is_robot_bounded(instructions: String) -> bool {
        
        let mut direction: i32 = 0;
        let mut x: i32 = 0;
        let mut y: i32 = 0;
        for ch in instructions.chars() {
            match ch {
                'G' => {
                    match direction {
                        0 => y += 1,
                        1 => x += 1,
                        2 => y -= 1,
                        3 => x -= 1,
                        _ => (),
                    }
                }
                'L' => direction = (direction + 3) % 4,
                'R' => direction = (direction + 1) % 4,
                _ => (),
            }
        }
        direction != 0 || (x == 0 && y == 0)
    }
}